### PR TITLE
Always check for root disks unless MINIO_CI_CD is set

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
The current code considers a pool with all root disks to be as part
of a testing environment even if there are other pools with mounted
disks. This will result to illegitimate writing in root disks.

Fix this by simplifing the logic: require MINIO_CI_CD in order to skip
root disk check.

## Motivation and Context
Avoid writing in root disks when a user wrongly mounts disks
of one given pool.

## How to test this PR?
Hard without faking some code

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
I did some improvements in this PR #14239

File: cmd/xl-storage.go [hunk 1]
```diff
@@ -212,27 +212,22 @@ func newXLStorage(ep Endpoint) (s *xlStorage, err error) {
...
-			}
 
-			// treat those disks with size less than or equal to the
-			// threshold as rootDisks.
-			rootDisk = info.Total <= globalRootDiskThreshold
-		}
+		// treat those disks with size less than or equal to the
+		// threshold as rootDisks.
+		rootDisk = info.Total <= globalRootDiskThreshold
 	}
```
Comment (harshavardhana) [inline cmd/xl-storage.go]: removing this part won't work check there is a slight optimization as well when globalRootDiskThreshold is provided we don't have to detect RootDisk code at all we can directly do GetInfo()

PTAL #14239

**Comment by harshavardhana** [comment]
PR https://github.com/minio/minio/pull/14239 already handles this part @vadmeste - I don't think this PR is needed.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/xl-storage.go [hunk 2]
```diff
@@ -212,31 +212,22 @@ func newXLStorage(ep Endpoint) (s *xlStorage, err error) {
...
+		// use - MINIO_ROOTDISK_THRESHOLD_SIZE to figure out if
+		// this disk is a root disk.
+		info, err := disk.GetInfo(path)
+		if err != nil {
+			return nil, err
 		}
+
+		// treat those disks with size less than or equal to the
+		// threshold as rootDisks.
+		rootDisk = info.Total <= globalRootDiskThreshold
```
Comment (harshavardhana) [inline cmd/xl-storage.go]: the problem with this approach is there are 'two' calls wasted when there is globalRootDiskThreshold set - the change that was done previously was to ensure that when `globalRootDiskThreshold` is set we have no reason to validate disk.IsRootDisk() or not. 

We simply have to just rely on either globalRootDiskThreshold or disk.IsRootDisk() - it is not required for disk.IsRootdisk() to be false to check globalRootDiskThreshold

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#d4d6f037-8e93-4d4d-a00e-6d2f3c9ce3bb) |
<a name="d4d6f037-8e93-4d4d-a00e-6d2f3c9ce3bb"></a>
### 14232-4e35ca1/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:30225
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 10 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 56 seconds
(4/14) Running aws-sdk-ruby tests ... done in 22 seconds
(5/14) Running awscli tests ... FAILED in 2 minutes and 25 seconds
{
  "name": "awscli",
  "duration": 91612,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-25981/awscli-mint-test-bucket-25981-objectremove_bucket failed: s3://awscli-mint-test-bucket-25981 An error occurred (SlowDown) when calling the DeleteBucket operation (reached max retries: 4): Resource requested is unreadable, please reduce your request rate"
}
(5/14) Running healthcheck tests ... done in 10 seconds
(6/14) Running mc tests ... FAILED in 11 seconds
{
  "name": "mc",
  "duration": "4999",
  "function": "test_rb",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-29708 --quiet --no-color mb myminio/mc-test-bucket-26195-1  >>> \nmc: <ERROR> Unable to make bucket `myminio/mc-test-bucket-26195-1`. Resource requested is unreadable, please reduce your request rate"
}
(6/14) Running minio-dotnet tests ... done in 1 minutes and 29 seconds
(7/14) Running minio-go tests ... done in 5 minutes and 15 seconds
(8/14) Running minio-java tests ... done in 3 minutes and 45 seconds
(9/14) Running minio-py tests ... FAILED in 10 minutes and 45 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-b8e9edd5-e2a9-4fec-a76d-c3185f72ef05",
    "object_name": "c9dbdc36-8e21-4f0e-a8c2-416bdda4b65e_0 ~ 0_1000"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-b8e9edd5-e2a9-4fec-a76d-c3185f72ef05, request_id: 16D220FBC93B769A, host_id: f496f8e3-5127-4ff6-81a9-e2898a8aabeb, bucket_name: minio-py-test-b8e9edd5-e2a9-4fec-a76d-c3185f72ef05",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 669, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-b8e9edd5-e2a9-4fec-a76d-c3185f72ef05, request_id: 16D220FBC93B769A, host_id: f496f8e3-5127-4ff6-81a9-e2898a8aabeb, bucket_name: minio-py-test-b8e9edd5-e2a9-4fec-a76d-c3185f72ef05\n",
  "duration": 589247
}
(9/14) Running s3cmd tests ... done in 1 minutes and 22 seconds
(10/14) Running s3select tests ... done in 2 minutes and 32 seconds
(11/14) Running versioning tests ... done in 6 minutes and 1 seconds

Executed 11 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |

All tests ran successfully
```
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**